rfd/0141-tsh-bench-db.md

```diff
@@ -0,0 +1,339 @@
+
+Having a flexible command like this enables developers to experiment with
+different scenarios besides relying on adding benchmark subcommands. For
+example, the same command can be used to measure multiple large amounts of
+connections constantly sending queries that return data (such as `SELECT 1;),
+but also can have a very complex query that returns a large chunk of data.
+
+After a general use, we can asses if more subcommands should be added to cover
+the most common use cases, making running benchmarks more convenient and less
+error-prone.
+
+```code
+tsh bench db query [--schema] [--db-user] [--uri=] [--connections=] [--create-sql=] [--query-sql=] [database]
+```
+
+| Flag            | Description |
+| --------------- | ----------- |
+| `--schema`      | If not provided the queries will be executed on the database's default schema. |
+| `--db-user`     | Database user used to connect to the target database. The user must have enough permissions on the database to execute all the benchmark queries. |
+| `--uri`         | Direct database access URI. When provided, the benchmark will issue connections directly to this database, and no Teleport is involved in the testing. It must contain all the connection information, including authentication credentials. The contents from --db-user will be ignored when provided. |
+| `--connections` | The number of connections used to run the benchmark queries. If not provided, the value will be the same as `--rate`. |
+| `--create-sql`  | Specifies literal contents or a path to a SQL file to create database tables and load them with data. |
+| `--query-sql`   | Specifies literal contents or a path to a SQL file that contains query statements to benchmark against the database. |
+| `database`      | Teleport target database name. Available databases can be retrieved by running `tsh db ls`. If not provided, the `--uri` flag (for direct database connection) must be present. | 
+
+The `--rate` common flag will indicate how many queries per second will be
+established. For example, if you provide the flags `--connections=1 --rate=25`,
+the benchmark will send 25 queries per second using the same connection.
+
+Note: `--create-sql` and `--query-sql` are followed by `-sql` suffixes because
+receiving custom queries in plain text format will only be supported on SQL
+databases initially. Other database protocols, such as MongoDB, won’t support
+this format as their query DSL is not plain text.
+
+### Direct database connections
+To directly connect to the databases, the suites must accept the credentials
+that will be used. To avoid adding multiple flags to cover the scenarios, we're
+heading into having a single `--uri` flag, which will receive the database URI
+with all options included. This way, users can choose the authentication method
+that will be used.
+
+### TLS routing support
+The benchmark commands will support TLS routing Teleport deployments by running
```

Comment:
I went with the "It would work the same way as `tsh db connect`": check if the database requires a local proxy and if it will be tunneled. But I don't see any problem with having a single behavior for connecting to databases.